src/utils/event-type-guards.ts

```diff
@@ -83,6 +83,18 @@ export function isSNSSQSEvent(event: any): event is SQSEvent {
   return false;
 }
 
+export function isEBSQSEvent(event: any): event is SQSEvent {
+  if (Array.isArray(event.Records) && event.Records.length > 0 && event.Records[0].eventSource === "aws:sqs") {
+    try {
+      const body = JSON.parse(event.Records[0].body) as EventBridgeEvent<any, any>;
+      return body["detail-type"] !== undefined;
```

Comment:
I wonder if we should be checking `detail` instead of `detail-type`, since we are getting the data from there.

I point this out since this can be an optional property.

Reply:
totally, I need to fix that in the other method as well - great point.

Reply:
Actually I think it's required, right? See `Minimum information needed for a valid custom event`:
https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events-structure.html

Reply:
I was guiding myself from https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-eventbridge/interfaces/puteventsrequestentry.html
<img width="1166" alt="Screenshot 2023-08-04 at 3 36 55 PM" src="https://github.com/DataDog/datadog-lambda-js/assets/30836115/77e850d1-71e0-4ff6-a94d-85f89c7d3b26">

But if we are really strict about it, `Detail` is also optional 😕

Reply:
I don't think those docs are correct, I can try it in the console and verify but I'm pretty sure you need both